src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/CapturedSymbol.cs

```diff
@@ -62,12 +62,13 @@ public override BoundExpression Replacement<TArg>(SyntaxNode node, Func<NamedTyp
         }
     }
 
-    internal sealed class CapturedToExpressionSymbolReplacement : CapturedSymbolReplacement
+    internal sealed class CapturedToExpressionSymbolReplacement<THoistedSymbolType> : CapturedSymbolReplacement
+        where THoistedSymbolType : Symbol
     {
         private readonly BoundExpression _replacement;
-        public readonly ImmutableArray<StateMachineFieldSymbol> HoistedFields;
+        public readonly ImmutableArray<THoistedSymbolType> HoistedFields;
```

Comment:
>HoistedFields [](http://example.com/codeflow?start=59&length=13)

This name seems out-of-date, as `THoistedSymbolType` can either be a `StateMachineFieldSymbol` or a `ParameterSymbol` or a `LocalSymbol`. 
Maybe we keep this field name and bring its type back to `ImmutableArray<StateMachineFieldSymbol>`, as the only usage of `HoistedFields` is for such fields, so I don't think we need to initialize this for runtime-async scenarios.

Reply:
I'd prefer to just update the name. The creation of these in `RefInitializationHoister` does not care about which scenario it was called for, and should remain generic IMO.